velox/functions/sparksql/aggregates/MinMaxByAggregate.cpp

```diff
@@ -44,13 +44,38 @@ struct SparkComparator {
     } else {
       if constexpr (sparkGreaterThan) {
         return !accumulator->hasValue() ||
-            (accumulator->compare(newComparisons, index) <= 0);
+            compare(
+                dynamic_cast<SingleValueAccumulator*>(accumulator),
+                newComparisons,
+                index) <= 0;
       } else {
         return !accumulator->hasValue() ||
-            (accumulator->compare(newComparisons, index) >= 0);
+            compare(
+                dynamic_cast<SingleValueAccumulator*>(accumulator),
+                newComparisons,
+                index) >= 0;
       }
     }
   }
+
+  FOLLY_ALWAYS_INLINE static int32_t compare(
+      const SingleValueAccumulator* accumulator,
+      const DecodedVector& decoded,
+      vector_size_t index) {
+    static const CompareFlags kCompareFlags{
+        true, // nullsFirst
+        true, // ascending
+        false, // equalsOnly
+        CompareFlags::NullHandlingMode::NoStop};
+
+    auto result = accumulator->compare(decoded, index, kCompareFlags);
+    VELOX_USER_CHECK(
```

Comment:
This check is redundant. NoStop will never return std::nullopt.

Reply:
Fixed